### PR TITLE
YLP-705 Hydrophone should consider caregivers as clinical accounts on pin-reset/forgot-emails and signup

### PR DESCRIPTION
**Pull request by CedricPerotto**
+ Using shoreline client from mdblp/shoreline (instead of go-common).

### CODE REVIEW

File: api/forgot.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
 
 	// if the resetter is already registered we can use his preferences
 	if resetUsr := a.findExistingUser(email, a.sl.TokenProvide()); resetUsr != nil {
-		if resetUsr.IsClinic() || a.Config.AllowPatientResetPassword {
+		if resetUsr.IsClinic() || resetUsr.HasRole("caregiver") || a.Config.AllowPatientResetPassword {
```
Comment (ydrisR) [inline api/forgot.go]: OR something like `!resetUsr.HasRole("patient") || a.Config.AllowPatientResetPassword` ?

File: api/pin_reset.go [hunk 2]
```diff
@@ -72,8 +72,8 @@ func (a *Api) SendPinReset(res http.ResponseWriter, req *http.Request, vars map[
 		return
 	}
 
-	if usrDetails.IsClinic() {
-		log.Printf("sendPinReset - Clinician account [%s] cannot receive PIN Reset message", usrDetails.UserID)
+	if usrDetails.IsClinic() || usrDetails.HasRole("caregiver") {
```
Comment (ydrisR) [inline api/pin_reset.go]:
```suggestion
	if !usrDetails.HasRole("patient") {
```

File: api/signup.go [hunk 3]
```diff
@@ -120,8 +120,8 @@ func (a *Api) sendSignUpInformation(res http.ResponseWriter, req *http.Request,
 		a.sendModelAsResWithStatus(res, status.StatusError{status.NewStatus(http.StatusInternalServerError, STATUS_ERR_FINDING_USER)}, http.StatusInternalServerError)
 		return
 	} else {
-		if usrDetails.IsClinic() {
-			log.Printf("Clinician account [%s] cannot receive information message", usrDetails.UserID)
+		if usrDetails.IsClinic() || usrDetails.HasRole("caregiver") {
```
Comment (ydrisR) [inline api/signup.go]:
```suggestion
		if !usrDetails.HasRole("patient") {
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,12 +3,13 @@
 Hydrophone is the module responsible for sending emails.
 This API sends notifications to users for things like forgotten passwords, initial signup, and invitations.
 
-## 1.6.0-rc1
+## 1.6.0
```
Comment (ydrisR) [inline CHANGELOG.md]: 1.6.1? And need a date

**Review by ydrisR — CHANGES_REQUESTED** [review]
a few suggetions, we can discuss as I'm not sure what is best :)

File: CHANGELOG.md (hunk 4 above)
Comment (CedricPerotto) [reply CHANGELOG.md]: > 1.6.1? And need a date

Version 1.6.0 wasn't released

**Review by ydrisR — APPROVED** [review]

**Pull request merged**